third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -1007,8 +673,14 @@ struct ConvertLayoutOpConversion
 } // namespace
 
 void mlir::triton::NVIDIA::populateConvertLayoutOpToLLVMPatterns(
-    LLVMTypeConverter &typeConverter, RewritePatternSet &patterns,
-    PatternBenefit benefit) {
-  patterns.add<ConvertLayoutOpConversion>(typeConverter, benefit);
-  patterns.add<LocalLoadOpConversion>(typeConverter, benefit);
+    LLVMTypeConverter &typeConverter, const TargetInfo &targetInfo,
+    RewritePatternSet &patterns, PatternBenefit benefit) {
+  // For now give ConvertLayoutOpConversion higher benefit, I can split before
+  // merging
+  patterns.add<ConvertLayoutOpConversion>(typeConverter, 100);
```

Comment:
remove hacks, move the generic pattern to outside this, split optimizations from functionality